[Hub Generated] Review request for Microsoft.KeyVault to add version preview/2022-02-01-preview

rahulalapati43 opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/dev-keyvault-Microsoft.KeyVault-2022-02-01-preview-swagger-completeness...main/).
## ARM API Information (Control Plane)


Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

rahulalapati43 committed:
Adds base for updating Microsoft.KeyVault from version stable/2022-07-01 to version 2022-02-01-preview

rahulalapati43 committed:
Updates readme

rahulalapati43 committed:
Updates API version in new specs and examples

rahulalapati43 committed:
Fixed breaking changes in managedHsm.json. (#22112)

Fixed LRO_RESPONSE_HEADER, INVALID_REQUEST_PARAMETER and OBJECT_MISSING_REQUIRED_PROPERTY errors.

Co-authored-by: Rahul Alapati <rahulalapati@microsoft.com>

rahulalapati43 commented:
/azp run unifiedPipeline

TimLovellSmith commented on specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json:
```diff
@@ -0,0 +1,2082 @@
+            "type": "string",
+            "enum": [
+              "2015-11-01",
+              "2022-02-01-preview"
```
>2022 [](http://example.com/codeflow?start=15&length=4)

This is puzzling me here.
this does not look like the usual way of modeling apiVersion. - as an enum not modeled as string. But then it also does not look like a keyVaults API per se - it seems to be an **ARM api**, So... does it really support this api version??

[https://learn.microsoft.com/en-us/rest/api/resources/resources/list](https://learn.microsoft.com/en-us/rest/api/resources/resources/list)

TimLovellSmith reviewed (changes requested):
:clock1:

rahulalapati43 committed:
Undid the API version changes to the Vaults_List API (#22142)

* Fixed breaking changes in managedHsm.json.
Fixed LRO_RESPONSE_HEADER, INVALID_REQUEST_PARAMETER and OBJECT_MISSING_REQUIRED_PROPERTY errors.

* Undid the API Version change to the VaultListFilterTypes API.

Co-authored-by: Rahul Alapati <rahulalapati@microsoft.com>
